rustls/src/conn/kernel.rs

```diff
@@ -0,0 +1,223 @@
+use core::marker::PhantomData;
+
+use alloc::boxed::Box;
+
+use crate::client::ClientConnectionData;
+use crate::common_state::Protocol;
+use crate::msgs::codec::{Codec, Reader};
+use crate::msgs::handshake::{CertificateChain, NewSessionTicketPayloadTls13};
+use crate::quic::Quic;
+use crate::server::ServerConnectionData;
+use crate::{CommonState, ConnectionTrafficSecrets, Error, ProtocolVersion, SupportedCipherSuite};
+
+/// A kernel client connection.
+///
+/// The only difference for kernel client vs server connections is that client
+/// connections have support for recording new session tickets.
+pub type KernelClientConnection = KernelConnection<ClientConnectionData>;
+
+/// A kernel server connection.
+///
+/// The only difference for kernel client vs server connections is that client
+/// connections have support for recording new session tickets.
+pub type KernelServerConnection = KernelConnection<ServerConnectionData>;
```

Comment:
I'd prefer to get rid of these type aliases -- I think these generally don't make code clearer but instead obfuscate it in more indirection.